libs/community/langchain_community/document_loaders/unstructured.py

```diff
@@ -151,20 +123,35 @@ class UnstructuredFileLoader(UnstructuredBaseLoader):
 
     References
     ----------
-    https://unstructured-io.github.io/unstructured/bricks.html#partition
+    https://docs.unstructured.io/open-source/core-functionality/partitioning
+    https://docs.unstructured.io/open-source/core-functionality/chunking
     """
 
     def __init__(
         self,
-        file_path: Union[str, List[str], Path, List[Path], None],
+        file_path: Union[str, list[str], Path, list[Path]],
+        *,
         mode: str = "single",
         **unstructured_kwargs: Any,
     ):
         """Initialize with file path."""
         self.file_path = file_path
+
+        try:
+            import unstructured  # noqa:F401
+        except ImportError:
+            raise ImportError(
+                "unstructured package not found, please install it with "
+                "`pip install unstructured`"
+            )
+
+        if not satisfies_min_unstructured_version("0.5.4"):
+            if "strategy" in unstructured_kwargs:
+                unstructured_kwargs.pop("strategy")
+
         super().__init__(mode=mode, **unstructured_kwargs)
 
-    def _get_elements(self) -> List:
+    def _get_elements(self) -> list:
```

Comment:
Why this change? Saw a similar one related to `Dict` above.

Reply:
Just following type-hinting practices I've learned from Steve.

Reply:
Lmk if you want me to revert these changes or if I can resolve these comments

Reply:
Actually, here's the PEP for deprecating `typing.List` and other generics. 
https://peps.python.org/pep-0585/#abstract
https://docs.python.org/3/library/typing.html#deprecated-aliases